lib/grammers-client/src/client/auth.rs

```diff
@@ -261,6 +265,58 @@ impl Client {
             Err(InvocationError::Rpc(RpcError { name, .. })) if name.starts_with("PHONE_CODE_") => {
                 Err(SignInError::InvalidCode)
             }
+            Err(InvocationError::Rpc(RpcError { name, .. }))
+                if name.starts_with("SESSION_PASSWORD_NEEDED") =>
+            {
+                Err(SignInError::PasswordRequired)
+            }
+            Err(error) => Err(SignInError::Other(error)),
+        }
+    }
+
+    pub async fn two_factor_auth(
```

Comment:
We may want to bikeshed on the naming here, and also we have to document this method with usage examples.

Reply:
I don't know about another name :/ For me, this one looks nice, but we can use `password_check` for example?

Reply:
I've named it `check_password` for now. This is how tg calls it. And looks like we've got no better variants in tg chat :/